app_dart/lib/src/request_handlers/scheduler/batch_backfiller.dart

```diff
@@ -99,15 +102,50 @@ class BatchBackfiller extends RequestHandler {
       });
       // Schedule all builds asynchronously.
       // Schedule after db updates to avoid duplicate scheduling when db update fails.
-      await Future.wait<void>(backfillRequestList(backfill));
+      await _scheduleWithRetries(backfill);
     } catch (error) {
       log.severe('Failed to update tasks when backfilling: $error');
     }
   }
 
+  /// Schedules tasks with retry when hitting pub/sub server errors.
+  Future<void> _scheduleWithRetries(List<Tuple<Target, FullTask, int>> backfill) async {
+    const RetryOptions retryOptions = Config.schedulerRetry;
+    try {
+      await retryOptions.retry(
```

Comment:
If we begin a backfill and get 90 percent through it where some were successful but then we hit a pub/sub error will it reattempt the successful tasks that were scheduled already?

Reply:
From `backfillRequestList`, you can see we are handling each task as a separate request. This means the failed request affects only the included target itself. So we do not have issue scheduling duplicated successful tasks.